components/channel_view/channel_view.jsx

```diff
@@ -70,7 +71,19 @@ export default class ChannelView extends React.PureComponent {
         return this.refs.channelView;
     }
 
+    componentDidUpdate(prevProps) {
+        if (prevProps.channelId !== this.props.channelId) {
+            mark('ChannelView#componentDidUpdate');
+
+            measure('SidebarChannelLink#click', 'ChannelView#componentDidUpdate');
+            measure('SidebarChannelLink#click', 'ChannelView#render');
```

Comment:
Because `measure` returns the measured duration in ms it's trivial for us to send the data to, say, Segment with:
```js
const duration = measure('SidebarChannelLink#click', 'ChannelView#componentDidUpdate');
trackEvent('ui', 'sidebar-channel-link-click-update', { duration });
```
... in fact I could change it to return the measure *and* name at some point and then do...
```js
const [duration, name] = measure('SidebarChannelLink#click', 'ChannelView#componentDidUpdate');
trackEvent('ui', name, { duration });
```